### PR TITLE
Fix: address minor form migration issues that could result in fatal errors

### PR DESCRIPTION
**Pull request by pauloiankoski**
## Description

During the testing of Form Migrations, we encountered errors caused by minor code details. This pull request aims to resolve those issues and ensure smooth Form Migrations.

This is a summary of the changes:
- Previously, if the attributes `customAmountMin` and `customAmountMax` were empty, they were saved as `0.0` post rounding implementation. Now, a validity check is performed on these attributes before rounding and saving them to ensure accurate data storage.
- Identified an issue where email notifications in P2P contained open-quotes, which interfered with form settings while extracting JSON content. This has been resolved by replacing open-quotes with double-quotes before setting the notifications in the migrated form.
- Noticed a migration step that saved the form prematurely before the migration was fully completed. This was unnecessary and had the potential to trigger unexpected side effects. This step has been rectified to ensure the form is saved only after the migration process is entirely finalized.

## Affects

Form Migrations

## Testing Instructions

1. Create a new P2P campaign that creates a new form for you.
2. Open that recently created form and migrate it to v3.
3. The form is migrated with no errors.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205692499999974

### CODE REVIEW

File: src/FormMigration/Steps/EmailSettings.php [hunk 1]
```diff
@@ -20,7 +20,11 @@ public function process()
                 'status' => $notification->get_notification_status($this->formV2->id),
                 'email_subject' => $notification->get_email_subject($this->formV2->id),
                 'email_header' => $notification->get_email_header($this->formV2->id),
-                'email_message' => $notification->get_email_message($this->formV2->id),
+                'email_message' => str_replace(
+                    ['"“', '“"', '“'],
```
Comment (glaubersilva) [inline src/FormMigration/Steps/EmailSettings.php:24]: This replacement is not working as expected. I migrated a P2P form and when opened it I got a fatal error that prevented the form preview from being rendered. Checking it deeper I realize that the JSON used in the Form Builder after the migration still keeps the same wrong quotes. 

What worked for me in my local tests was this replacement:

```
'email_message' => str_replace(
	['”', '”'],
	'"',
	$notification->get_email_message($this->formV2->id)
),
```

image

**Review by glaubersilva — CHANGES_REQUESTED** [review]
@pauloiankoski Nice work! Almost everything working as expected, I just left a comment about a minor detail related to the wrong quotes that still throws a fatal error.

File: src/FormMigration/Steps/EmailSettings.php (hunk 1 above)
Comment (pauloiankoski) [reply src/FormMigration/Steps/EmailSettings.php]: I've found that I wasn't including all possible variations for the quotes. Please, check it again. It must work as expected now.

**Review by glaubersilva — APPROVED** [review]
@pauloiankoski  Thanks for the update. Now it worked as expected, so looks good to QA.

**Review by rickalday — APPROVED** [review]
Passed manual QA tests

**Pull request merged**